acme/acme/client.py

```diff
@@ -259,12 +262,16 @@ def retry_after(cls, response, default):
         try:
             seconds = int(retry_after)
         except ValueError:
-            # pylint: disable=no-member
-            decoded = werkzeug.parse_date(retry_after)  # RFC1123
-            if decoded is None:
-                seconds = default
-            else:
-                return decoded
+            # The RFC 2822 parser handles all of RFC 2616's cases in modern
+            # environments (primarily HTTP 1.1+ but also py27+)
+            when = parsedate_tz(retry_after)
+            if when is not None:
+                try:
+                    tz_secs = datetime.timedelta(when[-1] if when[-1] else 0)
+                    return datetime.datetime(*when[:7]) - tz_secs
+                except (ValueError, OverflowError):
```

Comment:
`OverflowError` can be raised by either the `timedelta` call or the subtract operation on the `datetime` objects.  `ValueError` can only be raised by the second line, but I believe the code as written is cleaner than two nested try blocks.